src/__test__/redux/actions/projects/createProject.test.js

```diff
@@ -1,72 +1,108 @@
+import createProject from 'redux/actions/projects/createProject';
+import initialProjectsState from 'redux/reducers/projects';
+import { saveProject } from 'redux/actions/projects';
+import { createExperiment } from 'redux/actions/experiments';
+import {
+  PROJECTS_CREATE, PROJECTS_SAVING, PROJECTS_SAVED, PROJECTS_ERROR,
+} from 'redux/actionTypes/projects';
+
+// import endUserMessages from 'utils/endUserMessages';
 
 const mockStore = configureStore([thunk]);
 
-uuidv4.mockImplementation(() => 'random-project-uuid');
 jest.mock('uuid');
+const projectUuid = 'random-project-uuid';
+uuidv4.mockImplementation(() => projectUuid);
 
 jest.mock('../../../../redux/actions/projects/saveProject');
 saveProject.mockImplementation(() => async () => { });
 
 jest.mock('../../../../redux/actions/experiments/createExperiment');
-createExperiment.mockImplementation((projectUuid, projectName) => async () => ({
-  name: projectName,
-  uuid: projectUuid,
+createExperiment.mockImplementation((uuid, name) => async () => ({
+  name,
+  uuid,
 }));
 
+jest.mock('../../../../utils/pushNotificationMessage');
+pushNotificationMessage.mockImplementation(() => async () => { });
+
+enableFetchMocks();
+
 describe('createProject action', () => {
   beforeEach(() => {
     jest.clearAllMocks();
     jest.doMock();
+
+    jest.useFakeTimers('modern').setSystemTime(new Date('2020-01-01').getTime());
+
+    fetchMock.resetMocks();
+    fetchMock.doMock();
   });
 
-  const mockProject = {
-    ...initialState,
-    name: 'test project',
-    uuid: '12345',
-    description: 'Test project',
-    createdDate: '01-01-2021',
-    lastModified: '01-01-2021',
-  };
+  const projectName = 'test project';
+  const projectDescription = 'test project description';
+  const experimentName = 'mockExperimentName';
 
-  it('Dispatches event correctly', async () => {
+  it('Works correctly when there are no errors', async () => {
     const store = mockStore({
-      projects: {},
+      projects: initialProjectsState,
     });
-    await store.dispatch(createProject(mockProject.name, mockProject));
+
+    fetchMock.mockResponse(JSON.stringify({}), { url: 'mockedUrl', status: 200 });
+
+    await store.dispatch(
+      createProject(projectName, projectDescription, experimentName),
+    );
+
+    // Fetch call is made
+    const fetchMockFirstCall = fetchMock.mock.calls[0];
+
+    const { body: fetchBody, method: fetchMethod } = fetchMockFirstCall[1];
+    expect(fetchMockFirstCall[0]).toEqual(`http://localhost:3000/v1/projects/${projectUuid}`);
+
+    expect(fetchMethod).toEqual('POST');
+    expect(JSON.parse(fetchBody)).toMatchSnapshot();
+
+    expect(createExperiment).toHaveBeenCalledWith(projectUuid, experimentName);
 
     const actions = store.getActions();
 
-    // And then create and save projects
-    expect(actions[0].type).toEqual(PROJECTS_CREATE);
+    expect(actions[0].type).toEqual(PROJECTS_SAVING);
+    expect(actions[1].type).toEqual(PROJECTS_SAVED);
```

Comment:
No need to have the `PROJECTS_SAVED` be dispatched, just do the saved change in `PROJECTS_CREATE`